File: samples/java/Bingo/Server/Api/src/main/java/systems/zlink/samples/bingo/server/api/handlers/AuthenticatePlayerHandler.java
package systems.zlink.samples.bingo.server.api.handlers;

import systems.zlink.framework.ZLinkMessageContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.bingo.server.configuration.SampleNames;
import systems.zlink.samples.bingo.shared.contracts.BingoMessages;
import systems.zlink.samples.bingo.shared.contracts.Messages;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

@ZLinkHandlerGroup(SampleNames.ApiChannel)
public final class AuthenticatePlayerHandler
        implements ZLinkRequestHandler<
                Messages.AuthenticatePlayerReq, Messages.AuthenticatePlayerRes> {
    @Override
    public CompletionStage<Messages.AuthenticatePlayerRes> handle(
            Messages.AuthenticatePlayerReq request, ZLinkMessageContext context) {
        String actorId = request.getAccessToken();
        return CompletableFuture.completedFuture(
                BingoMessages.authenticatePlayerRes(true, actorId, actorId, null));
    }
}

File: samples/kotlin/Bingo/Server/Api/src/main/kotlin/systems/zlink/samples/kotlin/bingo/server/api/handlers/AuthenticatePlayerHandler.kt
package systems.zlink.samples.kotlin.bingo.server.api.handlers

import systems.zlink.framework.ZLinkMessageContext
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.framework.kotlin.ZLinkSuspendingRequestHandler
import systems.zlink.samples.kotlin.bingo.server.configuration.SampleNames
import systems.zlink.samples.kotlin.bingo.shared.contracts.AuthenticatePlayerReq
import systems.zlink.samples.kotlin.bingo.shared.contracts.AuthenticatePlayerRes

@ZLinkHandlerGroup(SampleNames.ApiChannel)
class AuthenticatePlayerHandler() :
    ZLinkSuspendingRequestHandler<AuthenticatePlayerReq, AuthenticatePlayerRes> {
    override suspend fun handle(request: AuthenticatePlayerReq, context: ZLinkMessageContext) =
        run {
            AuthenticatePlayerRes(
                accepted = true,
                actorId = request.accessToken,
                displayName = request.accessToken,
                reason = null,
            )
        }
}
